hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/handlers/HandlersChainFactory.java

```diff
@@ -51,7 +62,35 @@ public HttpHandler provide() {
                                                                 messageErrorProcessor, throughputLimiter);
         TopicHandler topicHandler = new TopicHandler(readHandler, topicsCache, messageErrorProcessor);
 
-        return topicHandler;
+        boolean authenticationEnabled = configFactory.getBooleanProperty(FRONTEND_AUTHENTICATION_ENABLED);
+        return authenticationEnabled ? withAuthenticationHandlersChain(topicHandler) : topicHandler;
+    }
+
+    private HttpHandler withAuthenticationHandlersChain(HttpHandler next) {
+        AuthenticationConfiguration authConfig = authenticationConfigurationProvider.getAuthenticationConfiguration()
+                .orElseThrow(() -> new IllegalStateException("AuthenticationConfiguration was not provided"));
+
+        try {
+            return createAuthenticationHandlersChain(next, authConfig);
+        } catch (Exception e) {
+            throw new IllegalStateException("Could not create authentication handlers chain", e);
```

Comment:
On the other hand `AuthenticationMechanism` has access to `HttpServerExchange` so it is possible to have a custom implementation logging this info.